src/lib.test.js

```diff
@@ -1,7 +1,96 @@
 import BrandgilityEmbeddedApi from './lib';
 
-test('instance should be an instance of BrandgilityEmbeddedApi', () => {
-  const brandgilityEmbeddedApi = new BrandgilityEmbeddedApi(window);
+let iframe;
+let brandgilityEmbeddedApi;
+let callback;
 
-  expect(brandgilityEmbeddedApi).toBeInstanceOf(BrandgilityEmbeddedApi);
+const identityKey = '@BrandgilityEmbeddedApi';
+const action = 'some action';
+
+beforeEach(() => {
+  callback = jest.fn();
+  iframe = document.createElement('iframe');
+  document.body.append(iframe);
+  brandgilityEmbeddedApi = new BrandgilityEmbeddedApi(iframe.contentWindow);
+});
+
+test('callback should be called when appropriate action has been received', () => {
+  brandgilityEmbeddedApi.on(action, callback);
+
+  const messageEvent = new MessageEvent('message', {
+    data: { identityKey, action },
+    source: iframe.contentWindow,
+  });
+
+  window.dispatchEvent(messageEvent);
+
+  expect(callback).toHaveBeenCalled();
+});
+
+test('callback should be called with payload sent with action', () => {
+  const payload = {
+    someField: 'someValue',
+  };
+
+  brandgilityEmbeddedApi.on(action, callback);
+
+  const messageEvent = new MessageEvent('message', {
+    data: { identityKey, action, payload },
+    source: iframe.contentWindow,
+  });
+
+  window.dispatchEvent(messageEvent);
+
+  expect(callback).toHaveBeenCalledWith(payload);
+});
+
+test('callback should NOT be called when received action from unexpected event source', () => {
+  const differentIframe = document.createElement('iframe');
+
+  document.body.append(differentIframe);
+
+  const messageEvent = new MessageEvent('message', {
+    data: { identityKey, action },
+    source: differentIframe.contentWindow,
+  });
+
+  window.dispatchEvent(messageEvent);
+
+  expect(callback).not.toHaveBeenCalled();
+});
```

Comment:
this test case is not complete, please listen for `action`
expectation is correct